models/ios/l3_interfaces/ios_l3_interface.yaml

```diff
@@ -1,88 +1,136 @@
+      name:
+        description:
+        - Full name of the interface excluding any logical unit number, i.e. GigabitEthernet0/1.
+        type: str
+        required: true
+      ipv4:
+        description:
+        - IPv4 address to be set for the Layer-3 interface mentioned in I(name) option.
+          The address format is <ipv4 address>/<mask>, the mask is number in range
+          0-32 eg. 192.168.0.1/24.
+        type: list
+        elements: dict
+        suboptions:
+          address:
+            description:
+            - Configures the IPv4 address for Interface.
+            type: str
+          secondary:
+            description:
+            - Configures the IP address as a secondary address.
+            type: bool
+          dhcp_client:
+            description:
+            - Configures and specifies client-id to use over DHCP ip. Note, This option
+              shall work only when dhcp is configured as IP.
+            - GigabitEthernet interface number
+            - This option is DEPRECATED and is replaced with dhcp which
+              accepts dict as input this attribute will be removed after 2023-08-01.
+            type: int
+          dhcp_hostname:
+            description:
+            - Configures and specifies value for hostname option over DHCP ip. Note,
+              This option shall work only when dhcp is configured as IP.
+            - This option is DEPRECATED and is replaced with dhcp which
+              accepts dict as input this attribute will be removed after 2023-08-01.
+            type: str
+          dhcp:
+            description: IP Address negotiated via DHCP.
+            type: dict
+            suboptions:
+              client_id:
+                description: Specify client-id to use.
+                type: str
+              hostname:
+                description: Specify value for hostname option.
+                type: str
+          pool:
+            description: IP Address auto-configured from a local DHCP pool.
+            type: str
+      ipv6:
+        description:
+        - IPv6 address to be set for the Layer-3 interface mentioned in I(name) option.
+        - The address format is <ipv6 address>/<mask>, the mask is number in range
+          0-128 eg. fd5d:12c9:2201:1::1/64
+        type: list
+        elements: dict
+        suboptions:
+          address:
+            description:
+            - Configures the IPv6 address for Interface.
+            type: str
+          autoconfig:
+            description: Obtain address using auto-configuration.
+            type: dict
+            suboptions:
+              default:
+                description: Insert default route.
+                type: bool
+          dhcp:
+            description: Obtain a ipv6 address using DHCP.
+            type: dict
+            suboptions:
+              rapid_commit:
+                description: Enable Rapid-Commit.
+                type: bool
+          ipv6_config:
```

Comment:
This doesn't seem right to me. If these are booleans, that is how they should be added in the model.